app/src/main/java/eu/kanade/tachiyomi/ui/history/HistoryScreenModel.kt

```diff
@@ -63,9 +63,9 @@ class HistoryScreenModel(
                 val beforeDate = before?.item?.readAt?.time?.toLocalDate() ?: LocalDate.MIN
                 val afterDate = after?.item?.readAt?.time?.toLocalDate() ?: LocalDate.MIN
                 when {
-                    beforeDate.isAfter(afterDate)
-                        or afterDate.equals(LocalDate.MIN)
-                        or beforeDate.equals(LocalDate.MIN) -> HistoryUiModel.Header(afterDate)
+                    beforeDate.equals(LocalDate.MIN)
+                        or beforeDate.isAfter(afterDate)
+                        and !afterDate.equals(LocalDate.MIN) -> HistoryUiModel.Header(afterDate)
```

Comment:
Maybe something like this? Alongside the other comment. Same for Updates screen
```suggestion
                    beforeDate != afterDate and  afterDate != null -> HistoryUiModel.Header(afterDate)
```

Reply:
Pushed the fix, it looks a lot cleaner. I think I got too hung up on wanting to use the date comparison functions.